src/platform/Linux/bluez/Helper.cpp

```diff
@@ -1197,20 +1197,21 @@ static void UpdateAdditionalDataCharacteristic(BluezGattCharacteristic1 * charac
     CHIP_ERROR err = CHIP_NO_ERROR;
     chip::System::PacketBufferHandle bufferHandle;
 
-    char serialNumber[ConfigurationManager::kMaxSerialNumberLength + 1];
+    uint8_t rotatingDeviceIdUniqueId[ConfigurationManager::kRotatingDeviceIDUniqueIDLength];
+    MutableByteSpan rotatingDeviceIdUniqueIdSpan(rotatingDeviceIdUniqueId);
     uint16_t lifetimeCounter = 0;
     BitFlags<AdditionalDataFields> additionalDataFields;
 
 #if CHIP_ENABLE_ROTATING_DEVICE_ID
-    err = ConfigurationMgr().GetSerialNumber(serialNumber, sizeof(serialNumber));
+    err = ConfigurationMgr().GetRotatingDeviceIdUniqueId(rotatingDeviceIdUniqueIdSpan);
     SuccessOrExit(err);
     err = ConfigurationMgr().GetLifetimeCounter(lifetimeCounter);
     SuccessOrExit(err);
 
     additionalDataFields.Set(AdditionalDataFields::RotatingDeviceId);
 #endif
 
-    err = AdditionalDataPayloadGenerator().generateAdditionalDataPayload(lifetimeCounter, serialNumber, strlen(serialNumber),
+    err = AdditionalDataPayloadGenerator().generateAdditionalDataPayload(lifetimeCounter, rotatingDeviceIdUniqueIdSpan,
```

Comment:
I am unsure here - this forces passing in of a rotation device id even if no rotating device id is enabled in the SDK.

ifdefind parameters sounds  ugly though. Maybe we should have a `class AdditionalDataPayloadParams` or something like that, which has a `SetRotatingUniqueId` which can be ifdefed.

Reply:
I tried to follow your suggestion and create AdditionalDataPayloadParams structure having rotating device id parameters (lifetimeCounter and uniqueId under the ifdef). The problem I have with that is that `AdditionalDataPayloadGenerator` doesn't see `CHIP_ENABLE_ROTATING_DEVICE_ID` definition. It cannot include `CHIPDeviceConfig.h`, as `additional_data_payload` module should has `src/platform` as a dependency, while `src/platform` already has `additional_data_payload` as a dependency ... I need to spent more time on digging in that, as I don't see for now easy solution to fix this.